source/language/openpulse.rst

```diff
@@ -0,0 +1,317 @@
+format:
+
+- It is more readable
+- Absolute time is handled through built-in timekeeping instructions
+- Gates and classical instructions can be mixed in with the pulses to create far richer calibrations
+- Pulse definitions are tied to circuit instructions rather than circuit programs
+- Richer handling of frames
+
+Channels
+--------
+
+Channels map to a hardware resource, which can play pulses to manipulate a qubit
+or capture signal from the qubit to perform a measurement. There is a many-to-many
+relationship between qubits and channels. One qubit may have multiple channels
+connecting to it. Pulses on different channels would have different physical
+interactions with that qubit. A channel may also have many qubits. For instance,
+a channel could manipulate the coupling between two neighboring qubits, or
+could even reference multiple qubits in a chain.
+
+Channels are vendor-specific and device-specific. It is expected that vendors
+of quantum hardware provide the appropriate channel names and qubit mappings
+as configuration information to end users. Each channel may also have associated 
+static settings, such as local-oscillator frequencies, which do not vary 
+throughout program execution. Again it is expected that vendors of quantum 
+hardware provide a method for manipulating those static settings if appropriate.
+
+There are two kinds of channels: transmit channels (sending input to a quantum
+device) and receive channels (reading output from a quantum device).
+
+Interacting with channels directly is onerous and makes the pulse programs less
+portable. The OpenPulse defcalgrammar does not directly work with channels and
+instead works with abstract "frames", described below. The static settings above 
+should also include a mapping of these abstract frames to the correct channels.
+
+Frames
+------
+
+It turns out to be quite useful to keep track of a set of carrier signals
+throughout the execution of a program. These carrier signals are called "frames"
+and are defined by :math:$A*e^{i\left(2\pi f t + \theta\right)}$, where `f` is
+frequency, `theta` is phase, and `A` is a scaling factor. Frames will also track
+time appropriately so programs do not need to deal in absolute time. The
+canonical motivation for keeping track of phase is to implement a "virtual
+Z-gate", which does not require a physical pulse but rather shifts the phase of
+all future pulses on that frame.
+
+The ``frame`` type is a compile-type type representing a *reference* to a frame.
+The frame is composed of four parts:
+
+1. A frequency ``frequency`` of type ``float``. It is not initialized and must
+be set before any pulses can be played on that frame.
+2. A phase ``phase`` of type ``angle``. This is initialized to the value 0.
+3. An amplitude ``scale`` of type ``float``. This is initilized to the value 1.0.
+4. A time of type ``dt`` which is manipulated implicitly and cannot be modified
```

Comment:
Should it be clarified that `dt` is the sample clock time? As done here: https://qiskit.github.io/openqasm/language/delays.html#length-and-stretch-types.